### PR TITLE
Check first name only for api/v1/participant-validation

### PR DESCRIPTION
**Pull request by asmega**
### Context

- Policy has given the go ahead to auto validate TRNs based on first name and not the whole name

### Changes proposed in this pull request

- `ParticipantValidationService` can now be passed a config hash to check first name only
- This feature is disabled by default and will check the whole name by default
- The api endpoint `api/v1/participant-validation` checks first name now

### Guidance to review

- none

### Testing

- See how to test in a review app below

### Review Checks
- [x] All pages have automated accessibility checks via cypress
- [x] All pages have visual tests via cypress + percy
- [x] All `School` queries are correctly scoped - `eligible` when they need to be

### How can I view this in a review app?

- Use existing `NPQRegistrationApiToken` or create a new one
- See data extract spreadsheet for test data
```
curl -H "Authorization: Bearer TOKEN" https://ecf-review-pr-791.london.cloudapps.digital/api/v1/participant-validation/TRN?full_name=FULL_NAME_WITH_WONRG_LAST_NAME&date_of_birth=DOB&nino=
```
- Should return record regardless of last names

### CODE REVIEW

File: app/services/participant_validation_service.rb [hunk 1]
```diff
@@ -44,8 +49,15 @@ def matching_record(trn:, nino:, full_name:, dob:)
     matches = 0
     trn_matches = padded_trn == dqt_record[:teacher_reference_number]
     matches += 1 if trn_matches
-    name_matches = full_name == dqt_record[:full_name]
+
+    name_matches = if check_first_name_only?
+                     full_name.split(" ").first.to_s == dqt_record[:full_name].split(" ").first.to_s
```
Comment (kicferk1) [inline app/services/participant_validation_service.rb:54]: I wonder, do we need `to_s` here? I would hope `split` would already give us a string.

I guess that makes sure neither value is `nil`, just not sure if it's worth it.

**Review by kicferk1 — APPROVED** [review]

File: app/services/participant_validation_service.rb (hunk 1 above)
Comment (asmega) [reply app/services/participant_validation_service.rb:54]: hmm i think i glossed over this so it looks more like a mistake. i will remove the `to_s`. If one or both values are `nil` it shouldn't affect `==`.

Comment (asmega) [reply app/services/participant_validation_service.rb]: FYI I've now removed the `to_s` in a new commit

**Review by kicferk1 — APPROVED** [review]

**Review by ethanmills — APPROVED** [review]
Thanks for sorting this. Just had confirmation that this applies to R&P too

**Comment by asmega** [comment]
Sorry for the delay. still waiting for sign off. If you need this in let me know then I can remove the controller change and add that line later and you will have the updated service

**Comment by ethanmills** [comment]
Blocked per Phil's comment ☝️

**Pull request merged**